engine/src/test/java/io/camunda/zeebe/engine/processing/incident/OutputMappingIncidentTest.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.processing.incident;
+
+import static io.camunda.zeebe.protocol.record.intent.IncidentIntent.CREATED;
+import static io.camunda.zeebe.protocol.record.intent.IncidentIntent.RESOLVED;
+import static java.util.Map.entry;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import io.camunda.zeebe.engine.util.EngineRule;
+import io.camunda.zeebe.model.bpmn.Bpmn;
+import io.camunda.zeebe.model.bpmn.BpmnModelInstance;
+import io.camunda.zeebe.protocol.record.Assertions;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.intent.IncidentIntent;
+import io.camunda.zeebe.protocol.record.intent.ProcessInstanceIntent;
+import io.camunda.zeebe.protocol.record.value.ErrorType;
+import io.camunda.zeebe.protocol.record.value.IncidentRecordValue;
+import io.camunda.zeebe.test.util.collection.Maps;
+import io.camunda.zeebe.test.util.record.ProcessInstances;
+import io.camunda.zeebe.test.util.record.RecordingExporter;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class OutputMappingIncidentTest {
+
+  @ClassRule public static final EngineRule ENGINE = EngineRule.singlePartition();
+  private static final String PROCESS_ID = "processId";
+  private final BpmnModelInstance bpmnModelInstance;
+  private final String elementId;
+  private final boolean createsJob;
+
+  public OutputMappingIncidentTest(
+      final BpmnModelInstance bpmnModelInstance, final String elementId, final boolean createsJob) {
+    this.bpmnModelInstance = bpmnModelInstance;
+    this.elementId = elementId;
+    this.createsJob = createsJob;
+  }
+
+  @Parameters(name = "{index}: {0}")
```

Comment:
:wrench: The name of the test case is not human readable. 

We could add a parameter for a description of the scenario.  

![Screenshot from 2021-09-14 13-52-44](https://user-images.githubusercontent.com/4305769/133253036-ce35b3cf-901e-4146-bc37-9140d87e898e.png)